jenkins/shared-library/vars/publish.groovy

```diff
@@ -0,0 +1,84 @@
+import java.nio.file.Paths
+
+void call(Map config, String phase) {
+	logger.logInfo("publishing data for ${phase}, ${config}")
+
+	publisher(config, phase)
+}
+
+String getResource(String fileName) {
+	return libraryResource(fileName)
+}
+
+void copyFileListToTemp(String[] files) {
+	files.each { filepath ->
+		fileHelper.copyToTempFile(getResource(Paths.get('artifacts').resolve(filepath).toString()), filepath)
+	}
+}
+
+void copyWithJenkinsFunctionFileToTemp(String[] files) {
+	copyFileListToTemp(files.plus(['scripts/jenkins-functions.sh']) as String[])
```

Comment:
(files + ['scripts/jenkins-functions.sh'])

Reply:
yea, the linter was complaining.  I am assuming its due to ``['scripts/jenkins-functions.sh']`` is an object array instead of String[]